cms/djangoapps/contentstore/signals/handlers.py

```diff
@@ -64,11 +64,13 @@ def listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=
         # Push the course outline to learning_sequences asynchronously.
         update_outline_from_modulestore_task.delay(course_key_str)
 
-    # Finally call into the course search subsystem
+    # Finally, call into the course search subsystem
     # to kick off an indexing action
     if CoursewareSearchIndexer.indexing_is_enabled() and CourseAboutSearchIndexer.indexing_is_enabled():
         update_search_index.delay(course_key_str, datetime.now(UTC).isoformat())
 
+    update_discussions_settings_from_course_task.delay(course_key_str)
```

Comment:
Does this need to be triggered every time? Or should it be triggered only if the course has in-context discussions enabled?

Reply:
I don't want to add that logic to a generic place. So that logic is in the task itself. Additionally, there may be other things that this code can do which apply to providers that don't support in-context discussions as well.